nmigen_boards/arty_s7.py

```diff
@@ -164,6 +164,7 @@ def toolchain_program(self, product, name, *, programmer="vivado", flash=True):
         assert programmer in ("vivado", "openocd")
 
         if programmer == "vivado":
+            assert not flash
```

Comment:
Something like this would be nicer:
```suggestion
            assert not flash, "Flash programming is not supported with Vivado yet"
```